src/ClientHandle.cpp

```diff
@@ -1395,187 +1395,166 @@ void cClientHandle::FinishDigAnimation()
 
 
 
-void cClientHandle::HandleRightClick(int a_BlockX, int a_BlockY, int a_BlockZ, eBlockFace a_BlockFace, int a_CursorX, int a_CursorY, int a_CursorZ, const cItem & a_HeldItem)
-{
-	// TODO: Rewrite this function
-
+void cClientHandle::HandleRightClick(int a_BlockX, int a_BlockY, int a_BlockZ, eBlockFace a_BlockFace, int a_CursorX, int a_CursorY, int a_CursorZ, int a_Hand)
+{
+	// This function handles three actions:
+	// (1) Place a block
+	// (2) "Use" a block: Interactive with the block, like opening a chest/crafting table/furnace
+	// (3) Use the held item targeting a block: e.g. farming
+	// (4) Use the held item without targeting a block: eating, drinking, charging a bow, using buckets
+	//
+	// As for 1.12.2, This function has two use cases:
+	// * Calling with parameters -1, 255, -1, BLOCK_FACE_NONE(-1), 0, 0, 0, Hand. --(4)
+	// * Calling with valid parameters.                                           --(1) (2) or (3)
+	//   Sneaking player will prioritize placement over using block, otherwise the other way round.
+	//
+	// In version 1.8.x, these two cases shareing the same packet id. In version >= 1.9, there is a new packet id "Use Item".
+	// But we still call this function in its handler for backward compatibility.
```

Comment:
Just use two functions(e.g. the current protocol design) and do the backward compability in the 1.8 Protocol handler (e.g. if packet == Case 4, then call the second function).

This has the benefit of reducing the function complexity, shrinking this comment, reduces client/server inconsistencies (the server thinks it is 1-3, but the client send 4) and if Cuberite should drop the 1.8 support, we don't have unnecessary "backward compability" code.

Reply:
Done